usbsdmux/usbsdmux.py

```diff
@@ -100,13 +144,104 @@ def mode_host(self, wait=True):
         """
         Switches the MicroSD-Card to the Host.
 
-        This Command will issue a disconnect first to make sure the the SD-card
-        has been properly disconnected from both sides and it's supply was off.
+        This Command will issue a disconnect first to make sure the SD-card
+        has been properly disconnected from both sides and its supply was off.
         """
         self.mode_disconnect(wait)
 
-        # the disconnect-command has already switched the card to the host.
-        # Thus we don't need to worry about glitches anymore.
+        # The disconnect-command has already switched the card to the host.
+        # Thus, we don't need to worry about glitches anymore.
 
         # now connect data and power
         self._pca.output_values(self._DAT_enable | self._PWR_enable | self._select_HOST | self._card_inserted)
+
+
+class UsbSdMuxFast:
+    """
+    Class to provide an interface for the multiplexer on an usb-sd-mux.
+    """
+
+    _DAT_enable = 0x00
+    _DAT_disable = Tca6408.gpio_2
+
+    _PWR_enable = 0x00
+    _PWR_disable = Tca6408.gpio_1
+
+    _select_DUT = Tca6408.gpio_0
+    _select_HOST = 0x00
+
+    _card_inserted = 0x00
+    _card_removed = Tca6408.gpio_3
+
+    # TODO: Add support for the GPIOs
+
+    def __init__(self, sg):
+        """
+        Create a new UsbSdMux.
+
+        Arguments:
+        sg -- /dev/sg* to use
+        """
+        self._tca = Tca6408(sg)
+
+    def get_mode(self):
+        """
+        Returns currently selected mode as string
+        """
+        val = self._tca.read_register(1)[0]
+
+        # If the SD-Card is disabled, we do not need to check for the selected mode.
+        # PWR_disable and DAT_disable are always switched at the same time.
+        # Let's assume it is sufficient to check one of both.
+        if val & self._PWR_disable:
+            return "off"
+
+        if val & self._select_DUT:
+            return "dut"
+        return "host"
+
+    def mode_disconnect(self, wait=True):
+        """
+        Will disconnect the Micro-SD Card from both host and DUT.
```

Comment:
Instead of duplicating the docstrings, put them on the base class.

Reply:
Added a proper base class and moved all comments there.